test/airport.spec.js

```diff
@@ -0,0 +1,331 @@
+
+// // Clean up - here we are clearing the variables ready for next test, to avoid contamination
+airport = null;
+plane = null;
+expected = undefined;
+actual = undefined;
+result = undefined;
+
+console.log(`============================`);
+
+//Test 10 - try to land a second instance of the same plane
+
+console.log(`============================`);
+console.log(`Test 10 - try to landPlane Plane(id) that is already in landedPlanes`);
+
+// Arrange - this is setting up the variables required for the test
+airport = new Airport(3); //set capacity higher than planes going in, so the capacity is not what blocks a re-landing plane
+plane1 = new Plane('Delta');
+expected = 1;
+
+// Act - this is running the methods to be tested and collecting the actual results
+airport.landPlane(plane1);
+airport.landPlane(plane1);
+actual = airport.getLandedPlaneCount(); //Boolean added to this function in the class, yes but in the end used getLandedPlaneCount!
+
+//Assert - testing the actual results against expected results
+result = assertEquals(expected, actual);
+console.log(`Test 10 - 'Plan cannot land if already landed': ${result}`);
+
+// // Clean up - here we are clearing the variables ready for next test, to avoid contamination
+airport = null;
+plane = null;
+expected = undefined;
+actual = undefined;
+result = undefined;
+
+console.log(`============================`);
+
+
+//Test 11 - test to check that a plane that has taken off, cannot take off again
+
+console.log(`============================`);
+console.log(`Test 11 - try to takeOffPlane Plane(id) that has already taken off`);
+
+// Arrange - this is setting up the variables required for the test
+airport = new Airport(1);
+plane1 = new Plane('Delta');
+expected = false;
+
+// Act - this is running the methods to be tested and collecting the actual results
+airport.landPlane(plane1); //land a plane at the airport first!
+//console.log(airport.isPlanePresent(plane1));
+airport.takeOffPlane(plane1);
+//console.log(airport.isPlanePresent(plane1));
+airport.takeOffPlane(plane1); 
+actual = airport.takeOffPlane(plane1); 
+
+//Assert - testing the actual results against expected results
+result = assertEquals(expected, actual);
+console.log(`Test 11 - 'Plan cannot take off a second time': ${result}`);
+
+// // Clean up - here we are clearing the variables ready for next test, to avoid contamination
+airport = null;
+plane = null;
+expected = undefined;
+actual = undefined;
+result = undefined;
+
+console.log(`============================`);
+
+//Test 12 - test to check that only a positive integer is accepted for capacity values
+
+console.log(`============================`);
+console.log(`Test 12 - 'try to set defaultAirportCapacity as a float or negative'`);
+
+// Arrange - this is setting up the variables required for the test
+defaultLandedCapacity = -4; //6.5 also tried
+airport = new Airport(defaultLandedCapacity);
+expected = 0; //would ideally throw an error but not sure how to do that
+
```

Comment:
defaultAirportCapacity becomes zero if a -ve or float is attempted, but would rather have a 'throw error'